### PR TITLE
Cleanup Atomic & Clump usage

### PR DESCRIPTION
**Pull request by danhedron**
- GameObjects now have either an Atomic or a Clump
- The transform of this object is updated to reflect where the object should be
- Accidentally fixed the player having the wrong model in the intro

### CODE REVIEW

File: rwengine/src/objects/GameObject.hpp [hunk 1]
```diff
@@ -92,16 +87,30 @@ class GameObject {
...
+        }
+        return NullAtomic;
+    }
+
+    const ClumpPtr& getClump() const {
+        if (auto clump = std::get_if<ClumpPtr>(&model_))
+        {
+            return *clump;
+        }
+        return NullClump;
```
Comment (ShFil119) [inline rwengine/src/objects/GameObject.hpp]: Maybe it'd better insert static instance of object into body of function? (above return)

File: rwengine/src/objects/GameObject.hpp [hunk 2]
```diff
@@ -92,16 +87,30 @@ class GameObject {
...
 
-    /**
-     * Changes the current model, used for re-dressing chars
-     */
+     const AtomicPtr& getAtomic() const {
+        if (auto atomic = std::get_if<AtomicPtr>(&model_))
+        {
+            return *atomic;
+        }
+        return NullAtomic;
```
Comment (ShFil119) [inline rwengine/src/objects/GameObject.hpp]: also

File: rwengine/src/objects/PickupObject.cpp [hunk 3]
```diff
@@ -271,6 +271,12 @@ PickupObject::PickupObject(GameWorld* world, const glm::vec3& position,
 
     setEnabled(true);
     setCollected(false);
+
+    auto modelData = getModelInfo<SimpleModelInfo>();
+    auto atomic = modelData->getAtomic(0)->clone();
+    atomic->setFrame(std::make_shared<ModelFrame>());
```
Comment (ShFil119) [inline rwengine/src/objects/PickupObject.cpp]: Maybe atomic should have method for this. Logic is "outside".

File: rwengine/src/objects/GameObject.hpp (hunk 1 above)
Comment (danhedron) [reply rwengine/src/objects/GameObject.hpp]: Putting a local static here will generate some guard instructions, unlike a static member. [example](https://godbolt.org/z/M3K8RB).

**Pull request merged**